### ISSUE TITLE
Unable to parse "[ 1 2 3 ]"

### ISSUE THREAD

**Issue by fhucho** [nicklockwood/Consumer#6]
Parsing `[ 1 2 3 ]` using this grammar:

```swift
let item: Consumer<String> = .character(in: "0" ... "9")
let array: Consumer<String> = ["[", .optional(" "), .interleaved(item, " "), .optional(" "), "]"]
print(try array.match("[ 1 2 3 ]"))
```

fails with:

```
Fatal error: Error raised at top level: Unexpected token ']' at 1:9 (expected '0' – '9')
```

**Comment by fhucho** [comment]
This works:

`let array: Consumer<String> = ["[", .interleaved(.optional(" "), item), "]"]`

**Comment by nicklockwood** [comment]
@fhucho fixed in 0.4.0. Thanks for reporting it!

**Comment by fhucho** [comment]
That was fast, thanks!
